pkg/kubelet/kuberuntime/utils.go

```diff
@@ -0,0 +1,349 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	kubecontainer "k8s.io/kubernetes/pkg/kubelet/container"
+)
+
+const (
+	// containerNamePrefix is used to identify the containers on the node managed by kubelet
+	containerNamePrefix = "k8s"
+
+	// Taken from lmctfy https://github.com/google/lmctfy/blob/master/lmctfy/controllers/cpu_controller.cc
+	minShares     = 2
+	sharesPerCPU  = 1024
+	milliCPUToCPU = 1000
+)
+
+// getKubeletPodSandboxes lists all pod sandboxes or just the running ones.
+func (m *kubeGenericRuntimeManager) getKubeletPodSandboxes(all bool) ([]*runtimeApi.PodSandbox, error) {
+	var resp []*runtimeApi.PodSandbox
+	var err error
+
+	if all {
+		resp, err = m.runtimeService.ListPodSandbox(nil)
+	} else {
+		readyState := runtimeApi.PodSandBoxState_READY
+		resp, err = m.runtimeService.ListPodSandbox(&runtimeApi.PodSandboxFilter{
+			State: &readyState,
+		})
+	}
+	if err != nil {
+		glog.Errorf("ListPodSandbox failed: %v", err)
+		return nil, err
+	}
+
+	result := []*runtimeApi.PodSandbox{}
+	for _, s := range resp {
+		if s.Name == nil || len(s.GetName()) == 0 {
+			continue
+		}
+
+		_, _, err = kubecontainer.ParsePodFullName(s.GetName())
+		if err != nil {
+			glog.V(3).Infof("%s sandbox %s is not managed by kubelet", m.runtimeType, s.GetName())
+			continue
+		}
+
+		result = append(result, s)
+	}
+
+	return result, nil
+}
+
+// getKubeletContainers lists all container or just the running ones.
```

Comment:
nit: expand the comment.
`getKubeletContainers lists all (or just the running) containers managed by kubelet.